src/main/java/pages/cartPage/ShoppingCartTable.java

```diff
@@ -0,0 +1,282 @@
+        }
+        WebElement table = locateElement(tableLocator);
+        List<WebElement> rows = table.findElements(By.tagName("tr"));
+
+        if (rowNumber > 0 && rowNumber < rows.size()) {
+            return rows.get(rowNumber - 1);
+        } else {
+            throw new IllegalArgumentException("Row number [" + rowNumber + "] is out of range");
+        }
+    }
+
+    private WebElement getCell(int rowNumber, int columnNumber) {
+        if (rowNumber == 1) {
+            throw new IllegalArgumentException("Row number 1 is a header row, not valid for this operation");
+        }
+        List<WebElement> productRows = driver.findElements(By.xpath(tableXpath + "//tr[.//td]"));
+
+        if (rowNumber > 0 && rowNumber <= productRows.size() + 1) {
+            WebElement targetRow = productRows.get(rowNumber - 2);
+            List<WebElement> cells = targetRow.findElements(By.tagName("td"));
+
+            if (columnNumber > 0 && columnNumber <= cells.size()) {
+                return cells.get(columnNumber - 1);
+            } else {
+                throw new IllegalArgumentException("Column number [" + columnNumber + "] is out of range");
+            }
+        } else {
+            throw new IllegalArgumentException("Row number [" + rowNumber + "] is out of range");
+        }
+    }
+
+    private String getCellContent(WebElement cell) {
+        if (hasQuantityInput(cell)) {
+            return getQuantityInputValue(cell);
+        } else if (hasImageItem(cell)) {
+            return getImageItemSrc(cell);
+        } else if (hasDeleteButton(cell)) {
+            return getDeleteButtonHref(cell);
+        } else {
+            return cell.getText();
+        }
+    }
+
+    private boolean hasQuantityInput(WebElement cell) {
+        return !cell.findElements(quantityItemInput).isEmpty();
+    }
+
+    private String getQuantityInputValue(WebElement cell) {
+        return cell.findElements(quantityItemInput).getFirst().getAttribute("value");
+    }
+
+    private boolean hasImageItem(WebElement cell) {
+        return !cell.findElements(By.xpath(".//img")).isEmpty();
+    }
+
+    private String getImageItemSrc(WebElement cell) {
+        return cell.findElements(By.xpath(".//img")).getFirst().getAttribute("src");
+    }
+
+    private boolean hasDeleteButton(WebElement cell) {
+        return !cell.findElements(deleteItemButton).isEmpty();
+    }
+
+    private String getDeleteButtonHref(WebElement cell) {
+        return cell.findElement(By.xpath("./a")).getAttribute("href");
+    }
+
+    public List<String> getRowContent(int rowNumber) {
+        WebElement row = getRow(rowNumber);
+        List<WebElement> cells = row.findElements(By.tagName("td"));
+
+        List<String> cellData = new ArrayList<>();
+        for (WebElement cell : cells) {
+            cellData.add(getCellContent(cell));
+        }
+        return cellData;
+    }
+
+    public List<String> getColumnContent(int columnNumber) {
```

Comment:
This can be part of tests not page object, you can just get all rows and use streams to get values that you need in given moment:)

Reply:
Out of date. Method removed. There are no longer any methods that get column content. All that is left are methods that retrieve row contents. 

Now tester writing the test has to take care of fetching the column, having a list of rows. Just as you wrote. Thanks